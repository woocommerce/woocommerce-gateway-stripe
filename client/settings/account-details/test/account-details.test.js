/** @format */
/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import AccountStatus from '..';

describe( 'AccountStatus', () => {
	const renderAccountStatus = ( accountStatus ) => {
		return render( <AccountStatus accountStatus={ accountStatus } /> );
	};

	test( 'renders connected account', () => {
		const { container: accountStatus } = renderAccountStatus( {
			status: 'complete',
			paymentsEnabled: true,
			depositsStatus: 'daily',
			currentDeadline: 0,
			accountLink: '',
		} );
		expect( accountStatus ).toMatchSnapshot();
	} );

	test( 'renders restricted soon account', () => {
		const { container: accountStatus } = renderAccountStatus( {
			status: 'restricted_soon',
			paymentsEnabled: true,
			depositsStatus: 'daily',
			currentDeadline: 1583844589,
			accountLink:
				'/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments&wcpay-login=1',
		} );
		expect( accountStatus ).toMatchSnapshot();
	} );

	test( 'renders restricted account with overdue requirements', () => {
		const accountStatus = renderAccountStatus( {
			status: 'restricted',
			paymentsEnabled: false,
			depositsStatus: 'disabled',
			currentDeadline: 1583844589,
			pastDue: true,
			accountLink:
				'/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments&wcpay-login=1',
		} );
		expect( accountStatus ).toMatchSnapshot();
	} );

	test( 'renders restricted account', () => {
		const accountStatus = renderAccountStatus( {
			status: 'restricted',
			paymentsEnabled: false,
			depositsStatus: 'disabled',
			currentDeadline: 1583844589,
			pastDue: false,
			accountLink: '',
		} );
		expect( accountStatus ).toMatchSnapshot();
	} );

	test( 'renders rejected.other account', () => {
		const { container: accountStatus } = renderAccountStatus( {
			status: 'rejected.other',
			paymentsEnabled: false,
			depositsStatus: 'disabled',
			currentDeadline: 0,
			accountLink: '',
		} );
		expect( accountStatus ).toMatchSnapshot();
	} );

	test( 'renders rejected.fraud account', () => {
		const { container: accountStatus } = renderAccountStatus( {
			status: 'rejected.fraud',
			paymentsEnabled: false,
			depositsStatus: 'disabled',
			currentDeadline: 0,
			accountLink: '',
		} );
		expect( accountStatus ).toMatchSnapshot();
	} );

	test( 'renders rejected.terms_of_service account', () => {
		const { container: accountStatus } = renderAccountStatus( {
			status: 'rejected.terms_of_service',
			paymentsEnabled: false,
			depositsStatus: 'disabled',
			currentDeadline: 0,
			accountLink: '',
		} );
		expect( accountStatus ).toMatchSnapshot();
	} );

	test( 'renders manual (suspended) deposits', () => {
		const { container: accountStatus } = renderAccountStatus( {
			status: 'complete',
			paymentsEnabled: true,
			depositsStatus: 'manual',
			currentDeadline: 0,
			accountLink: '',
		} );
		expect( accountStatus ).toMatchSnapshot();
	} );
} );
