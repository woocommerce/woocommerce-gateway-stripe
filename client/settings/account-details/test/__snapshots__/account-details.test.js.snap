// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountStatus renders connected account 1`] = `
<div>
  <div>
    <div>
      <span
        class="chip chip-primary is-compact"
      >
        Complete
      </span>
      <span
        class="account-status__info"
      >
        Payments:
        <span
          class="account-status__info__green"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-yes-alt"
          />
          Enabled
        </span>
      </span>
      <span
        class="account-status__info"
      >
        Deposits:
        <span
          class="account-status__info__green"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-yes-alt"
          />
          Daily
        </span>
      </span>
    </div>
    
  </div>
</div>
`;

exports[`AccountStatus renders manual (suspended) deposits 1`] = `
<div>
  <div>
    <div>
      <span
        class="chip chip-primary is-compact"
      >
        Complete
      </span>
      <span
        class="account-status__info"
      >
        Payments:
        <span
          class="account-status__info__green"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-yes-alt"
          />
          Enabled
        </span>
      </span>
      <span
        class="account-status__info"
      >
        Deposits:
        <span
          class="account-status__info__yellow"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-warning"
          />
          Temporarily suspended (
          <a
            href="https://docs.woocommerce.com/document/payments/faq/deposits-suspended/"
            rel="noopener noreferrer"
            target="_blank"
          >
            learn more
          </a>
          )
        </span>
      </span>
    </div>
    
  </div>
</div>
`;

exports[`AccountStatus renders rejected.fraud account 1`] = `
<div>
  <div>
    <div>
      <span
        class="chip chip-light is-compact"
      >
        Rejected
      </span>
      <span
        class="account-status__info"
      >
        Payments:
        <span
          class="account-status__info__red"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-warning"
          />
          Disabled
        </span>
      </span>
      <span
        class="account-status__info"
      >
        Deposits:
        <span
          class="account-status__info__red"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-warning"
          />
          Disabled
        </span>
      </span>
    </div>
    <div
      class="account-status__desc"
    >
      This account has been rejected because of suspected fraudulent activity.
    </div>
  </div>
</div>
`;

exports[`AccountStatus renders rejected.other account 1`] = `
<div>
  <div>
    <div>
      <span
        class="chip chip-light is-compact"
      >
        Rejected
      </span>
      <span
        class="account-status__info"
      >
        Payments:
        <span
          class="account-status__info__red"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-warning"
          />
          Disabled
        </span>
      </span>
      <span
        class="account-status__info"
      >
        Deposits:
        <span
          class="account-status__info__red"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-warning"
          />
          Disabled
        </span>
      </span>
    </div>
    <div
      class="account-status__desc"
    >
      This account has been rejected.
    </div>
  </div>
</div>
`;

exports[`AccountStatus renders rejected.terms_of_service account 1`] = `
<div>
  <div>
    <div>
      <span
        class="chip chip-light is-compact"
      >
        Rejected
      </span>
      <span
        class="account-status__info"
      >
        Payments:
        <span
          class="account-status__info__red"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-warning"
          />
          Disabled
        </span>
      </span>
      <span
        class="account-status__info"
      >
        Deposits:
        <span
          class="account-status__info__red"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-warning"
          />
          Disabled
        </span>
      </span>
    </div>
    <div
      class="account-status__desc"
    >
      This account has been rejected due to a Terms of Service violation.
    </div>
  </div>
</div>
`;

exports[`AccountStatus renders restricted account 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div>
        <div>
          <span
            class="chip chip-alert is-compact"
          >
            Restricted
          </span>
          <span
            class="account-status__info"
          >
            Payments:
            <span
              class="account-status__info__red"
            >
              <span
                class="dashicon dashicons dashicons-dashicons-warning"
              />
              Disabled
            </span>
          </span>
          <span
            class="account-status__info"
          >
            Deposits:
            <span
              class="account-status__info__red"
            >
              <span
                class="dashicon dashicons dashicons-dashicons-warning"
              />
              Disabled
            </span>
          </span>
        </div>
        <div
          class="account-status__desc"
        >
          Payments and deposits are disabled for this account until business information is verified by the payment processor.
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <span
          class="chip chip-alert is-compact"
        >
          Restricted
        </span>
        <span
          class="account-status__info"
        >
          Payments:
          <span
            class="account-status__info__red"
          >
            <span
              class="dashicon dashicons dashicons-dashicons-warning"
            />
            Disabled
          </span>
        </span>
        <span
          class="account-status__info"
        >
          Deposits:
          <span
            class="account-status__info__red"
          >
            <span
              class="dashicon dashicons dashicons-dashicons-warning"
            />
            Disabled
          </span>
        </span>
      </div>
      <div
        class="account-status__desc"
      >
        Payments and deposits are disabled for this account until business information is verified by the payment processor.
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AccountStatus renders restricted account with overdue requirements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div>
        <div>
          <span
            class="chip chip-alert is-compact"
          >
            Restricted
          </span>
          <span
            class="account-status__info"
          >
            Payments:
            <span
              class="account-status__info__red"
            >
              <span
                class="dashicon dashicons dashicons-dashicons-warning"
              />
              Disabled
            </span>
          </span>
          <span
            class="account-status__info"
          >
            Deposits:
            <span
              class="account-status__info__red"
            >
              <span
                class="dashicon dashicons dashicons-dashicons-warning"
              />
              Disabled
            </span>
          </span>
        </div>
        <div
          class="account-status__desc"
        >
          Payments and deposits are disabled for this account until missing business information is updated. 
          <a
            href="/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments&wcpay-login=1"
          >
            Update now
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <span
          class="chip chip-alert is-compact"
        >
          Restricted
        </span>
        <span
          class="account-status__info"
        >
          Payments:
          <span
            class="account-status__info__red"
          >
            <span
              class="dashicon dashicons dashicons-dashicons-warning"
            />
            Disabled
          </span>
        </span>
        <span
          class="account-status__info"
        >
          Deposits:
          <span
            class="account-status__info__red"
          >
            <span
              class="dashicon dashicons dashicons-dashicons-warning"
            />
            Disabled
          </span>
        </span>
      </div>
      <div
        class="account-status__desc"
      >
        Payments and deposits are disabled for this account until missing business information is updated. 
        <a
          href="/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments&wcpay-login=1"
        >
          Update now
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AccountStatus renders restricted soon account 1`] = `
<div>
  <div>
    <div>
      <span
        class="chip chip-warning is-compact"
      >
        Restricted soon
      </span>
      <span
        class="account-status__info"
      >
        Payments:
        <span
          class="account-status__info__green"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-yes-alt"
          />
          Enabled
        </span>
      </span>
      <span
        class="account-status__info"
      >
        Deposits:
        <span
          class="account-status__info__green"
        >
          <span
            class="dashicon dashicons dashicons-dashicons-yes-alt"
          />
          Daily
        </span>
      </span>
    </div>
    <div
      class="account-status__desc"
    >
      To avoid disrupting deposits, 
      <a
        href="/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments&wcpay-login=1"
      >
        update this account
      </a>
       by 8am Mar 10, 2020 with more information about the business.
    </div>
  </div>
</div>
`;
